velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -207,6 +208,58 @@ struct UnixTimestampParseWithFormatFunction
   bool invalidFormat_{false};
 };
 
+/// Converts date string to Timestmap type.
+template <typename T>
+struct GetTimestampFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* /*input*/,
+      const arg_type<Varchar>* format) {
+    auto sessionTimezoneName = config.sessionTimezone();
+    if (!sessionTimezoneName.empty()) {
+      sessionTimezoneId_ = util::getTimeZoneID(sessionTimezoneName);
+    }
+    if (format != nullptr) {
+      this->formatter_ = buildJodaDateTimeFormatter(
+          std::string_view(format->data(), format->size()));
+      isConstantTimeFormat_ = true;
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Timestamp>& result,
+      const arg_type<Varchar>& input,
+      const arg_type<Varchar>& format) {
+    if (!isConstantTimeFormat_) {
```

Comment:
Does Spark allow non-constant format argument? Seems expensive to create this per row.

Reply:
Yes, from spark source code, I note non-constant format is allowed.